ls1intum/Artemis
# `Programming exercises`: Fix missing grouping of feedback

# Client tests are currently failing, as develop branch does not have enough line coverage



### Checklist
#### General



### Motivation and Context


#5923 added feedback groups. This PR fixes that certain feedbacks were not grouped correctly and therefore not visible to the viewer.

### Description

Feedback items get grouped by a predicate `shouldContain` defined in `FeedbackGroup`.
For programming-exercises we used the attribute `FeedbackItem.credits` to sort them, however that field is not always present so we switch to the attribute `FeedbackItem.positive`, which contains the same information but should always be present

### Steps for Testing

Prerequisites:
Go to a feedback detail view, look into the response of `details` and check if each feedback is visible in the view

### Review Progress






#### Code Review

### Test Coverage





### Screenshots
